File: Algorithms_python/38_Count_and_Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        s = '1'
        for _ in range(n-1):
            i = 0
            t = []
            while i < len(s):
                j = i
                while j < len(s) and s[j] == s[i]:
                    j += 1
                t.append(str(j - i))
                t.append(str(s[i]))
                i = j
            s = ''.join(t)
        return s
'''
垃圾题目，方法是设计一个计数器一个一个添加元素，对着添加，一个1两个1这种
'''

File: Algorithms_C#/38_Count_and_Say.cs
public class Solution {
    public string CountAndSay(int n) {
        string prev = "1";
        for (int i = 2; i <= n; ++i){
            StringBuilder curr = new StringBuilder();
            int start = 0, pos = 0;

            while(pos < prev.Length){
                while(pos < prev.Length && prev[pos] == prev[start]){
                    pos++;
                }
                curr.Append(pos - start);
                curr.Append(prev[start]);
                start = pos;
            }
            prev = curr.ToString();
        }
        return prev;
    }
}